### DISCUSSION TITLE
cursor shape and shell integrations

### DISCUSSION THREAD

**Discussion by leholehes** [kovidgoyal/kitty#4073] [General]
Hi

When shell integration is enabled it overwrites the cursor shape from the configuration file without any warning.
This behaviour is not intuitive at all. I spent half a day troubleshooting why my cursor is not as it was configured :-(

Why is it neccessary for shell integration to change cursor shape when there is already generic configuration option for this?
Can this behaviour be changed, because I would really like to try the new features but I also want my cursor to stay like I configured it at first place...

--
Leho

**Comment by page-down** [comment]
Perhaps you would like to configure the following option.

```
# no-cursor: turn off changing of the text cursor to a bar when editing text
shell_integration no-cursor
```
